### PR TITLE
Added Support for Feedback for Clashing Events

### LINKED ISSUE #176: Feedback when event clashes with one another

### PR DESCRIPTION
**Pull request by SorataYuu**
closes #176

### CODE REVIEW

File: src/main/java/cs2103/v15_1j/jimjim/model/DataLists.java [hunk 1]
```diff
@@ -124,7 +105,8 @@ public void add(TaskEvent taskEvent) {
 			Collections.sort(eventsList);
 		}
 	}
-	
+
+	//@@author Jeremy
```
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/model/DataLists.java]: I thought u are suppose to put your matric no. not your name?

File: src/main/java/cs2103/v15_1j/jimjim/ui/MainViewController.java [hunk 2]
```diff
@@ -183,5 +187,18 @@ public void showFatalError(String message){
...
 	}
+	
+	public void addEvent(Event event){
+		boolean clashes = dayPickerPaneController.addEvent(event);
+		
+		if(!clashes){
+			showNotification("\""+event.getName() + "\" has been added.");
+		}
+		else {
+			showNotification("\""+event.getName() + "\" clashes with another event!");
```
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/ui/MainViewController.java]: I think it's better to name one event that this event clashes with

Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/ui/MainViewController.java]: Also these kind of messages should be in feedback, not here.

File: src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java [hunk 3]
```diff
@@ -73,17 +74,36 @@ public void refreshData(){
...
+			}
+
+			rowFactory.addLabel("", "");
 		}
 
 		showFloatingTasks();
 	}
 
 	private void showFloatingTasks(){
+		rowFactory.addLabel("Floating Tasks", "header");
```
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java]: I don't think we need a header "Floating Tasks". Just display them without a date and it should be clear enough

File: src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java [hunk 4]
```diff
@@ -73,17 +74,36 @@ public void refreshData(){
...
 		hasCompleted = rowFactory.showAllFloatingTasks(showCompleted);
 
 		int index = todayGridPane.getChildren().size();
 
+		if(displayList.getFloatingTasksList().size() == 0){
```
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java]: I don't understand this part

File: src/main/java/cs2103/v15_1j/jimjim/ui/MainViewController.java (hunk 2 above)
Comment (SorataYuu) [inline src/main/java/cs2103/v15_1j/jimjim/ui/MainViewController.java]: Hmm, but there's no point in getting the ViewCon to create a feedback that would then execute a method in ViewCon :/

File: src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java (hunk 3 above)
Comment (SorataYuu) [inline src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java]: Hmm, Initially I didn't put the header. But I realize that it is not as obvious, I think it looks better with the header.

File: src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java (hunk 4 above)
Comment (SorataYuu) [inline src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java]: If there are no floating tasks, display a message

File: src/main/java/cs2103/v15_1j/jimjim/ui/MainViewController.java (hunk 2 above)
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/ui/MainViewController.java]: this function will be executed by AddFeedback when you add an event? If that's the case just move those logic there.

File: src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java (hunk 4 above)
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java]: Well personally I don't seeing a lot of "No x/y/z" messages like this. If there is such a message at all, it should be when there are absolutely nothing at all.

File: src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java (hunk 3 above)
Comment (tran-tien-dat) [inline src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java]: Most Todo list program don't have header for tasks without time right? I don't think the header helps the user in any way.

Comment (SorataYuu) [inline src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java]: Depends on the program, most usually don't put them together. Plus we're putting Today, Overdue and Floating together. It'll get very confusing for the user if it's not clear.

File: src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java (hunk 4 above)
Comment (SorataYuu) [inline src/main/java/cs2103/v15_1j/jimjim/ui/TodayPaneController.java]: I think it'll help, like Henry said feedback is useful and that it's not No news is good news. In this case it's only when there are completely no floating tasks and I think it's fine to just include that for now.

**Comment by tran-tien-dat** [comment]
Please merge if you are not adding anything else

**Pull request merged**